tests/Phinx/Db/Adapter/PostgresAdapterTest.php

```diff
@@ -833,6 +833,51 @@ public function testChangeColumn()
         $this->assertTrue($this->adapter->hasColumn('t', 'column2'));
     }
 
+    public function testChangeColumnFromTextToSmallInteger()
+    {
+        $table = new \Phinx\Db\Table('t', [], $this->adapter);
+        $table->addColumn('column1', 'text')
+            ->insert(['column1' => '32767'])
+            ->save();
+
+        $table->changeColumn('column1', 'smallinteger')->save();
+        $columnType = $table->getColumn('column1')->getType();
+        $this->assertTrue($columnType === 'smallinteger');
+
+        $row = $this->adapter->fetchRow('SELECT * FROM t');
+        $this->assertEquals(32767, $row['column1']);
```

Comment:
```suggestion
        $this->assertSame(32767, $row['column1']);
```

Reply:
This might not work in all DBs, as some return this as string I think.